src/vhdl/vhdl-canon.adb

```diff
@@ -80,6 +84,30 @@ package body Vhdl.Canon is
      (Conc_Stmt : Iir; Proc : Iir; Parent : Iir; Clear : Boolean) return Iir;
    procedure Canon_Conditional_Signal_Assignment_Expression (Stmt : Iir);
 
+   procedure Canon_Check_Edge_Func(Node : Iir) is
+      Func_Identifier : Name_Id;
+   begin
+      if not Canoning_Process then
+         return;
+      end if;
+
+      case Get_Kind (Node) is
+      when Iir_Kind_Function_Call =>
+         Func_Identifier := Get_Identifier (Get_Implementation (Node));
+         if Name_Table.Image (Func_Identifier) = "rising_edge" or
+            Name_Table.Image (Func_Identifier) = "falling_edge"
```

Comment:
Technically this is not correct.  A user can define its own function `rising_edge` that does something differently (OK, that doesn't happen).
You can check if the function is `rising_edge` using code like this: `Get_Implicit_Definition(Imp) = Iir_Predefined_Ieee_1164_Rising_Edge`.
You will certainly use a case statement to handle all the functions needed.

Reply:
I thought of that, and I though that user would most probably not do that so I went with the naive solution first.
I will modify it to use the "Get_Implicit_Defintion" approach.